content/shmem_reductions.tex

```diff
@@ -366,16 +366,40 @@ \subsubsubsection{PROD}
     when \CONST{\_\_STDC\_NO\_COMPLEX\_\_} is defined to 1}, an \openshmem
     implementation is not required to provide support for these
     complex-typed interfaces.
+
+    The binary operations performed by \openshmem reductions are intended to be
+    associative and commutative.
+    However, floating point arithmetic is not associative or commutative due to the inherent
+    inaccuracies of floating-point representations caused by rounding errors and
+    finite precision.
+    This can lead to variations in the result of \openshmem arithmetic reduction operations
+    on floating-point datatypes.
+
+    Each \ac{PE} in the team must produce a \dest{} array that is bitwise
+    identical to the \dest{} arrays produced by all other \acp{PE} in the team.
+
+    The result of any given \openshmem reduction is reproducible across job
+    execution environments.
+    In other words, repetition of the same reduction operation on the same
+    input \source{} array, same team, and same overall job configuration must
+    yield bitwise identical results across all \dest{} arrays of the \acp{PE}
+    in the team.
```

Comment:
Feedback from @stewartl318 - this may not be within the user's control (e.g., router tables, adaptive routing, congestion control, etc.) so it's probably not worth mentioning in a spec...  This observation might be too theoretical.

Could we simply say something like, "the result of reductions may vary across jobs (due to the possibility of various arithmetic orderings)"?  This is particularly true for min/max with NaNs...  should that be called out?

Reply:
More WG feedback (thanks @wokuno) - perhaps in the future, we can add something like a `reductions_are_ordered` field to the `shmem_team_config_t` for users to request a consistent ordering of the binary ops within reductions.

Reply:
Run-to-run results reproducibility is achievable, but it typically comes with a performance cost. Do OpenSHMEM users want to pay for that?

Reply:
Removing this statement for now.